dashboard/app/assets/javascripts/editor/json_editor.js

```diff
@@ -0,0 +1,106 @@
+/* global CodeMirror */
+
+function calloutEditor(container, options) {
+
+  container = $(container);
+
+  var callout_editor = CodeMirror.fromTextArea(container.find(options.json_textarea).get(0), {
+    mode: 'javascript',
+    viewportMargin: Infinity,
+    matchBrackets: true
+  });
+
+  // Create callout spaces for each callout in the original JSON
+  var calloutJSON = callout_editor.getValue();
+  if (calloutJSON.length > 0) {
+    var callouts_to_update = JSON.parse(callout_editor.getValue());
+    $.each(callouts_to_update, function (index, callout) {
+      updateTemplate(callout, $createNewCalloutSpace());
+    });
+  }
+
+  /**
+    * For each key in the given model, set the <input> with a matching class name to the key's value.
+    * @param {model} The model to use when updating the DOM.
+    * @param {$template} The jQuery element to search for <input> elements.
+    */
+  function updateTemplate(model, $template) {
+    $.each(model, function (key, value) {
+      if (value && typeof value === 'object') {
```

Comment:
Note that all of the code in here was simply moved from inline in the editors/all template to this standalone file. I modified it minimally to get it to work with some flexible parameters, but otherwise left it as is